### PR TITLE
Docs: Update Tutorial.rst

### LINKED ISSUE #2359: Prerequisites for coala
Since coala can run analysis on ANY language, there are some prerequisites that some users may not have if they are not python programmers.

Creating a seperate page that can be linked to on pages like [GettingStarted/Testing](https://coala.readthedocs.io/en/latest/Getting_Involved/Testing.html) or [Install](https://coala.readthedocs.io/en/latest/Users/Install.html) can provide support for successfully running coala from ground 0 without cluttering up the begining of the instructions for the 80% of users that already have  things like python3 and pip installed. 

> Prerequisite - coala requires python 3 https://www.python.org/download/releases/3.0/. You will need to make sure that is installed
> 
> Check what version you have installed with `python -V`
> 
> To run tests, you first need to install some dependencies. This can be done by following these steps:
> 
> **Download the repository**
> ```
> $ git clone https://github.com/coala-analyzer/coala.git
> $ # Use https://github.com/coala-analyzer/coala-bears.git for coala bears
> ```
>  
> **move to the coala directory** 
> ```
> $ cd coala  # cd coala-bears when testing bears
> $ # Use pip for python 3 on windows
> 
> ```
> **pip these files that should install everything you need for the tests**
> `$ pip3 install -r test-requirements.txt -r requirements.txt`
> 
> Let us know if you get any errors

or 

```
Setting up a Development Environment
====================================

In order to help develop coala, you will need the following prerequisites:

- Python 3.3 or higher installed on your PC (Check with ``python --version``)
- Clone the repository:
  ``git clone https://github.com/coala-analyzer/coala.git``, or
  ``git clone https://github.com/coala-analyzer/coala-bears.git`` if you want
  to develop the bears.
- Install the requirements:
  ``cd coala; pip install -r requirements.txt -r test-requirements.txt``, or cd
  into ``coala-bears`` respectively. Be sure that your pip really uses Python
  3, you might need to use ``pip3`` on some linux distributions.
```

### PR DESCRIPTION
**Pull request by jschwarzwalder**
Adding information to the Prerequisites about needing python3

Fixes #2359  https://github.com/coala-analyzer/coala/issues/2359

### CODE REVIEW

File: docs/Users/Tutorials/Tutorial.rst [hunk 1]
```diff
@@ -7,9 +7,12 @@ to the usage of *coala*.
 Prerequisites
 -------------
 
-In order to complete this tutorial you will need *coala* installed.
-(Installation is actually not required but it's more convenient and
-recommended.)
+In order to complete this tutorial you will need
+`python3.3 <https://www.python.org/download/releases/3.0/>`_ or higher
+installed on your PC (Check with ``python --version``) installed.
```
Comment (Adrianzatreanu) [inline docs/Users/Tutorials/Tutorial.rst]: ``installed on your PC installed`` should remove one of them

**Comment by Adrianzatreanu** [comment]
Same commit message problem as with your other 2 PRs (see https://github.com/coala-analyzer/coala/pull/2380 for more info). hint:
``Tutorial.rst: Change X``
``newline``
``Adds information to the prerequisites needing Python3.``
This is an example of a better commit message at least :) good luck!

File: docs/Getting_Involved/Testing.rst [hunk 2]
```diff
@@ -9,12 +9,25 @@ Executing our Tests
 *coala* has a big test suite. It is meant to work on every platform on
 every PC. If you just execute our tests you are doing us a favor.
 
-To run tests, you first need to install some dependencies. This can be
-done by executing:
+To run tests, you will need to open a file located in the default
+coala directory
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: end with period?

File: docs/Getting_Involved/Testing.rst [hunk 3]
```diff
@@ -9,12 +9,25 @@ Executing our Tests
...
+
+If you have not already, you can start by cloneing a copy of the project
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: cloning* not cloneing

File: docs/Getting_Involved/Testing.rst [hunk 4]
```diff
@@ -9,12 +9,25 @@ Executing our Tests
...
 
-    $ pip3 install -r test-requirements.txt
+    $ git clone https://github.com/coala-analyzer/coala.git
+
+Then navigate to the directory where coala is located
+
+Next you need to install some dependencies. This can be
+done by executing:
+
+    $ pip3 install -r test-requirements.txt  -r requirements.txt
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: thats not how you enclose it in code style, check the example at lines 21-23. first ``::`` and then newline and then indent your code.

File: docs/Getting_Involved/Testing.rst [hunk 5]
```diff
@@ -9,12 +9,25 @@ Executing our Tests
...
+
+Either fork the repo on Github https://github.com/coala-analyzer/coala,
+or clone it directly from coala
 
 ::
 
-    $ pip3 install -r test-requirements.txt
+    $ git clone https://github.com/coala-analyzer/coala.git
+
+Then navigate to the directory where coala is located
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: missing a period here too.

File: docs/Getting_Involved/Testing.rst [hunk 6]
```diff
@@ -9,12 +9,25 @@ Executing our Tests
...
-To run tests, you first need to install some dependencies. This can be
-done by executing:
+To run tests, you will need to open a file located in the default
+coala directory
+
+If you have not already, you can start by cloneing a copy of the project
+to your local machine.
+
+Either fork the repo on Github https://github.com/coala-analyzer/coala,
+or clone it directly from coala
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: maybe semicolon at end of ``coala``?

File: docs/Getting_Involved/Testing.rst [hunk 7]
```diff
@@ -9,12 +9,25 @@ Executing our Tests
...
 
 ::
 
-    $ pip3 install -r test-requirements.txt
+    $ git clone https://github.com/coala-analyzer/coala.git
+
+Then navigate to the directory where coala is located
+
+Next you need to install some dependencies. This can be
+done by executing:
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: ``done by executing  the following command while in the root of the coala project`` may be more helpful

**Comment by Adrianzatreanu** [comment]
unack f1f8854

**Comment by Adrianzatreanu** [comment]
hey it seems you pushed again but you ignored my review, can you please check it next time?

File: docs/Getting_Involved/Testing.rst [hunk 8]
```diff
@@ -9,12 +9,28 @@ Executing our Tests
 *coala* has a big test suite. It is meant to work on every platform on
 every PC. If you just execute our tests you are doing us a favor.
 
-To run tests, you first need to install some dependencies. This can be
-done by executing:
+To run tests, you will need to open a file located in the default
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: open what file? this sentence is so unclear to me, it's more unhelpful than helpful, which shouldnt be the case in docs

Comment (jschwarzwalder) [inline docs/Getting_Involved/Testing.rst]: I copied this phrase from a comment given to me on a previous issue I encountered. I can change the language but I worry it won't be accepted.

File: docs/Getting_Involved/Testing.rst [hunk 9]
```diff
@@ -9,12 +9,28 @@ Executing our Tests
...
-To run tests, you first need to install some dependencies. This can be
-done by executing:
+To run tests, you first need to install some dependencies.
+This can be done by following these steps:
+
+If you have not already, you can start by cloning a copy of the project
+to your local machine.
+
+Either fork the repo on Github https://github.com/coala-analyzer/coala,
+or clone it directly from coala:
```
Comment (hypothesist) [inline docs/Getting_Involved/Testing.rst]: Well we're still cloning it from GitHub, not from coala. Also, they'd need to clone it nevertheless - you can't run tests on GitHub :P

So IMO I'd rather change the sentence like `You can clone the repository (or a fork of it) by running:` with `repository` having a link to our page.

File: docs/Getting_Involved/Testing.rst (hunk 7 above)
Comment (hypothesist) [inline docs/Getting_Involved/Testing.rst]: Not really IMO, we've already asked them to navigate to the directory ^^

File: docs/Getting_Involved/Testing.rst [hunk 10]
```diff
@@ -9,12 +9,26 @@ Executing our Tests
 *coala* has a big test suite. It is meant to work on every platform on
 every PC. If you just execute our tests you are doing us a favor.
 
-To run tests, you first need to install some dependencies. This can be
-done by executing:
+To run tests, you first need to install some dependencies.
+This can be done by following these steps:
+
+If you have not already, You can clone the
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: two things: one) You shouldnt be with a capital Y
two) when you give directions, or steps, it should be in the imperivate mood. So:
``If you have not already, clone the..``

File: docs/Getting_Involved/Testing.rst [hunk 11]
```diff
@@ -9,12 +9,26 @@ Executing our Tests
...
+`repository <https://github.com/coala-analyzer/coala>`_ (or a fork of
+it) by running:
+
+::
+
+    $ git clone https://github.com/coala-analyzer/coala.git
+
+Then navigate to the directory where coala is located.
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: As earlier, these are steps, you should imperative it: ``Navigate to the..`` - no ``then``

File: docs/Getting_Involved/Testing.rst [hunk 12]
```diff
@@ -9,12 +9,26 @@ Executing our Tests
...
+
+Next you need to install some requirements. This can be
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: here the ``Next`` is okay, no need to change it, fits good.

File: docs/Getting_Involved/Testing.rst [hunk 13]
```diff
@@ -9,12 +9,26 @@ Executing our Tests
...
+done by executing the following command while in the root of the
+coala project.
```
Comment (Adrianzatreanu) [inline docs/Getting_Involved/Testing.rst]: maybe ``coala project directory`` is more accurate?

**Comment by Adrianzatreanu** [comment]
unack f52a45f

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2378/files/0308064cebd710944acf22526be546e271eb67db?short_path=cec1e94#diff-cec1e94e94756a6fa8c906a49ea3c245

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**